src/main/java/spoon/support/JavaOutputProcessor.java

```diff
@@ -195,17 +175,58 @@ public void createJavaFile(CtType<?> element) {
 
 	}
 
+	@Override
+	public boolean isToBeProcessed(CtNamedElement candidate) {
+		return candidate instanceof CtType<?> || candidate instanceof CtPackage && (candidate.getComments().size() > 0 || candidate.getAnnotations().size() > 0);
+	}
+
 	/**
 	 * Creates a source file for each processed top-level type and pretty prints
 	 * its contents.
 	 */
-	public void process(CtType<?> type) {
-		if (type.isTopLevel()) {
-			createJavaFile(type);
+	public void process(CtNamedElement nameElement) {
+		if (nameElement instanceof CtType && ((CtType) nameElement).isTopLevel()) {
+			createJavaFile((CtType<?>) nameElement);
+		} else if (nameElement instanceof CtPackage) {
+			createPackageFile((CtPackage) nameElement);
 		}
 		printer.reset();
 	}
 
+	private void createPackageFile(CtPackage pack) {
+		// create package directory
+		File packageDir;
+		if (pack.isUnnamedPackage()) {
```

Comment:
This code is duplicate of https://github.com/INRIA/spoon/pull/766/files#diff-291edfebee2e5667f55882f29dfec128R143.